lib/puppet/type/file/source.rb

```diff
@@ -340,7 +340,7 @@ def chunk_file_from_source(&block)
 
     def handle_response_error(response)
       message = "Error #{response.code} on SERVER: #{response.body.empty? ? response.reason : response.body}"
-      raise Net::HTTPError.new(message, response.nethttp)
+      raise Net::HTTPError.new(message, Puppet::HTTP::ResponseConverter.to_ruby_response(response))
```

Comment:
We really shouldn't be raising `Net::HTTPError` within puppet anymore, but that's a different issue than what this PR is fixing.